multinet/db.py

```diff
@@ -301,6 +323,31 @@ def workspace_table_keys(
     return keys
 
 
+# TODO: Ensure current user has writer role
+def create_workspace_table_from_aql(workspace: str, name: str, aql: str) -> str:
+    """Create a new workspace from an aql query."""
+    db = get_workspace_db(workspace, readonly=True)
+
+    if db.has_collection(name):
+        raise AlreadyExists("table", name)
+
+    # In the future, the result of this validation can be
+    # used to determine dependencies in virtual tables
+    try:
+        db.aql.validate(aql)
+        rows = list(db.aql.execute(aql))
+    except (AQLQueryValidateError, AQLQueryExecuteError) as e:
+        raise MalformedRequestBody(str(e))
```

Comment:
This doesn't seem right--it's not that the request body is bad, it's that the AQL query was bad. We probably need a specific 400-series error for this.

Reply:
The AQL query is sent in the request body, so it seems fitting to me :man_shrugging:. But if you feel strongly about creating a new error, I'm not opposed to it.

Reply:
This is a case where a user would get "malformed body" as an error, think oh I must have sent something in the wrong shape, then dig around, then get annoyed when they find out it's actually a syntax error in their AQL query. Since we already know it's a more specific error, it makes sense to send back a more specific error condition.

Does the validation step reveal syntax errors? and what kind of errors occur at execution time? I think having a couple of AQL-based error classes (in the 400-series) makes sense here.

Reply:
Yeah I agree, like having separate aql validation and execution errors. The validation step does reveal syntax, which is what is passed into the raised error as `str(e)`

Reply:
Lmk what you think of fd8f5be